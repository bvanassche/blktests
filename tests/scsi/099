#!/bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright 2022 Google LLC
#
# Test support for segment sizes less than 4096 bytes.

. tests/block/rc
. common/scsi_debug

DESCRIPTION="do IO on scsi_debug with 512 byte segments"
TIMED=1

requires() {
	_have_fio
	_have_scsi_debug
	_have_sub_page_limit_support
}

test() {
	local bs=4096

	echo "Running ${TEST_NAME}"
	local scsi_debug_params=(
		add_host=1
		clustering=1
		delay=0
		sector_size="$bs"      # bytes
	)
	if _have_module_param scsi_debug max_segment_size; then
		scsi_debug_params+=(max_segment_size=512) # bytes
		echo "Segment size = 512" >>"$FULL"
	fi
	if ! _init_scsi_debug "${scsi_debug_params[@]}"; then
		echo "Initializing scsi_debug failed"
		return 1
	fi
	local blkdev=/dev/${SCSI_DEBUG_DEVICES[0]}
	[ -b "$blkdev" ] || return 1
	local ioengine
	for ioengine in psync sg; do
		echo "$ioengine"
		fio --verify=md5 --rw=randwrite --bs=$bs --ioengine=$ioengine \
		    --thread --group_reporting --sync=1 --direct=1 \
		    --name=scsi-099 --filename="$blkdev" \
		    --output="${RESULTS_DIR}/block/fio-output-scsi-099-$ioengine.txt" \
		    >>"$FULL"
		local fio_status=$?
		[ $fio_status = 0 ] || break
	done
	_exit_scsi_debug
	case $fio_status in
		0) echo "Passed";;
		*) echo "Failed (fio status = $fio_status)";;
	esac
}
