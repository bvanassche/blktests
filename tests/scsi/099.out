Running scsi/099
psync
sg
Passed
